benchmarks/low_level.cpp

```diff
@@ -0,0 +1,455 @@
+        cbdc::transaction::wallet wallet_b;
+        auto mint_tx_a = wallet_a.mint_new_coins(4, 50);
+        wallet_a.confirm_transaction(mint_tx_a);
+        ASSERT_EQ(wallet_a.count(), 4);
+        ASSERT_EQ(wallet_b.count(), 0);
+
+        state.ResumeTiming();
+        m_valid_tx
+            = wallet_a.send_to(175, wallet_b.generate_key(), true).value();
+        state.PauseTiming();
+        wallet_a.confirm_transaction(m_valid_tx);
+        ASSERT_EQ(wallet_a.balance(), 25);
+        ASSERT_EQ(wallet_a.count(), 1);
+
+        wallet_b.confirm_transaction(m_valid_tx);
+        ASSERT_EQ(wallet_b.balance(), 175);
+        ASSERT_EQ(wallet_b.count(), 1);
+    }
+}
+
+// serialize full tx
+BENCHMARK_F(low_level, serialize_tx)(benchmark::State& state) {
+    // if(state.thread_index() == 0) {
+    //     GTEST_LOG_(INFO) << "Starting transaction operational benchmarks ... ";
+    // }
+    // serialize 1-1 tx
+    m_valid_tx = wallet1.send_to(100, wallet2.generate_key(), true).value();
+    for(auto _ : state) {
+        m_os << m_valid_tx;
+    }
+    m_of.close();
+}
+
+// serialize compact tx
+BENCHMARK_F(low_level, serialize_compact_tx)(benchmark::State& state) {
+    // serialize 1-1 tx
+    m_valid_tx = wallet1.send_to(100, wallet2.generate_key(), true).value();
+    auto cp_tx = cbdc::transaction::compact_tx(m_valid_tx);
+    for(auto _ : state) {
+        m_os << cp_tx;
+    }
+    m_of.close();
+}
+
+// deserialize full tx
+BENCHMARK_F(low_level, deserialize_tx)(benchmark::State& state) {
+    // serialize 1-1 tx
+    auto read_tx = cbdc::transaction::full_tx();
+    m_valid_tx = wallet1.send_to(100, wallet2.generate_key(), true).value();
+    for(auto _ : state) {
+        state.PauseTiming();
+        m_os.reset();
+        m_os << m_valid_tx;
+        m_of.close();
+        m_if.open(m_BENCHMARK_File);
+
+        read_tx = cbdc::transaction::full_tx();
+        m_is.reset();
+        state.ResumeTiming();
+        m_is >> read_tx;
+        state.PauseTiming();
+        ASSERT_EQ(read_tx, m_valid_tx);
+    }
+
+    m_if.close();
+}
+
+// deserialize compact tx
+BENCHMARK_F(low_level, deserialize_compact_tx)(benchmark::State& state) {
+    // serialize 1-1 tx
+    auto read_tx = cbdc::transaction::full_tx();
+    auto read_cp = cbdc::transaction::compact_tx(read_tx);
+
+    m_valid_tx = wallet1.send_to(100, wallet2.generate_key(), true).value();
+    auto cp_tx = cbdc::transaction::compact_tx(m_valid_tx);
+    for(auto _ : state) {
+        state.PauseTiming();
+        m_os.reset();
+        m_os << cp_tx;
+        m_of.close();
+        m_if.open(m_BENCHMARK_File);
+
+        read_cp = cbdc::transaction::compact_tx(read_tx);
+        m_is.reset();
+        state.ResumeTiming();
+        m_is >> read_cp;
+        state.PauseTiming();
+        ASSERT_EQ(cp_tx, read_cp);
+    }
+
+    m_if.close();
+}
+
+// roundtrip benchmarking full tx
+BENCHMARK_F(low_level, roundtrip_tx)(benchmark::State& state) {
```

Comment:
I love roundtrip tests, but since we already benchmark serialize and deserialize (because the individual performance of those is actually more helpful for us in most cases), I don't know that round-tripping here offers us much help beyond the two more granular cases.